Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/BoundsControl.cs

```diff
@@ -431,15 +449,36 @@ private void Awake()
             if (targetObject == null)
                 targetObject = gameObject;
 
+            // ensure we have a default configuration in case there's none set by the user
+            scaleHandlesConfiguration = EnsureScriptable(scaleHandlesConfiguration);
+            rotationHandlesConfiguration = EnsureScriptable(rotationHandlesConfiguration);
+            boxDisplayConfiguration = EnsureScriptable(boxDisplayConfiguration);
+            linksConfiguration = EnsureScriptable(linksConfiguration);
+            handleProximityEffectConfiguration = EnsureScriptable(handleProximityEffectConfiguration);
+
+            // instantiate runtime classes for visuals
+            scaleHandles = new BoundsControlScaleHandles(scaleHandlesConfiguration);
+            rotationHandles = new BoundsControlRotationHandles(rotationHandlesConfiguration);
+            boxDisplay = new BoundsControlBoxDisplay(boxDisplayConfiguration);
+            links = new BoundsControlLinks(linksConfiguration);
+            proximityEffect = new ProximityEffect(handleProximityEffectConfiguration);
+
             // subscribe to visual changes
-            scaleHandles.configurationChanged.AddListener(CreateRig);
-            scaleHandles.visibilityChanged.AddListener(ResetVisuals);
-            rotationHandles.configurationChanged.AddListener(CreateRig);
-            boxDisplay.configurationChanged.AddListener(CreateRig);
-            links.configurationChanged.AddListener(CreateRig);
-
-            // subscribe visuals to bounds changes
-            boundsUpdated.AddListener(() => rotationHandles.CalculateEdgeCenters(ref boundsCorners));
+            scaleHandlesConfiguration.configurationChanged.AddListener(CreateRig);
+            scaleHandlesConfiguration.visibilityChanged.AddListener(ResetVisuals);
+            rotationHandlesConfiguration.configurationChanged.AddListener(CreateRig);
+            boxDisplayConfiguration.configurationChanged.AddListener(CreateRig);
+            linksConfiguration.configurationChanged.AddListener(CreateRig);          
+        }
+
+        private static T EnsureScriptable<T>(T instance) where T : ScriptableObject
```

Comment:
Should this not be in extensions?

Reply:
i don't really want to put it into that extensions util as it's in the editor namespace and is just edit mode utils in there. this method is used during runtime. i could create a util class for runtime utils?

Reply:
ah right, dont' worry about it fornow